src/main/java/edu/kit/provideq/toolbox/format/cnf/dimacs/DimacsCNF.java

```diff
@@ -0,0 +1,151 @@
+package edu.kit.provideq.toolbox.format.cnf.dimacs;
+
+import com.bpodgursky.jbool_expressions.Expression;
+import com.bpodgursky.jbool_expressions.parsers.ExprParser;
+import com.bpodgursky.jbool_expressions.rules.RuleSet;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public class DimacsCNF {
+    final static String LINE_SEPARATOR = System.getProperty("line.separator");
+    final static char SEPARATOR = ' ';
+    final static char NEGATION_PREFIX = '-';
+    final static char COMMENT_START = 'c';
+    final static char PREAMBLE_START = 'p';
+    final static char CLAUSE_END = '0';
+    final static String CNF_IDENTIFIER = "cnf";
+
+    private final List<Variable> variables;
+    private final ArrayList<ArrayList<Variable>> orClauses;
+
+    public DimacsCNF(Expression<String> expression) {
+        this(new ExpressionToDimacsCNF().parse(expression));
+    }
+
+    public DimacsCNF(DimacsCNF cnf) {
+        this.orClauses = new ArrayList<>(cnf.orClauses);
+        this.variables = new ArrayList<>(cnf.variables);
+    }
+
+    public DimacsCNF(ArrayList<ArrayList<Variable>> orClauses) {
+        this(orClauses, orClauses.stream()
+                .flatMap(Collection::stream)
+                .distinct()
+                .toList());
+    }
+
+    public DimacsCNF(ArrayList<ArrayList<Variable>> orClauses, List<Variable> variables) {
+        this.variables = variables;
+        this.orClauses = orClauses;
```

Comment:
It would be nice to make copies of these lists but I can understand if this seems infeasible here